src/astro/databases/aws/redshift.py

```diff
@@ -113,4 +127,100 @@ def is_native_load_file_available(
+        and if it does, it transfers it and returns true else false.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        :param if_exists: Overwrite file if exists. Default False
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+        """
+        method_name = self.NATIVE_PATHS.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                native_support_kwargs=native_support_kwargs,
+                **kwargs,
+            )
+        else:
+            raise DatabaseCustomError(
+                f"No transfer performed since there is no optimised path "
+                f"for {source_file.location.location_type} to bigquery."
+            )
+
+    def load_s3_file_to_table(
+        self,
+        source_file: File,
+        target_table: Table,
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        Load content of multiple files in S3 to output_table in Redshift by:
+        - Creating a table
+        - Using the COPY command
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the redshift
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+
+        .. seealso::
+            `Redshift official documentation on CREATE TABLE
+            <https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html>`_
+            `Redshift official documentation on COPY
+            <https://docs.aws.amazon.com/redshift/latest/dg/r_COPY.html>`_
+        """
+        native_support_kwargs = native_support_kwargs or {}
+
+        table_name = self.get_table_qualified_name(target_table)
+        file_type = NATIVE_PATHS_SUPPORTED_FILE_TYPES.get(source_file.type.name)
+
+        iam_role = (
+            native_support_kwargs.pop("IAM_ROLE")
+            if native_support_kwargs.get("IAM_ROLE")
+            else None
+        )
+        copy_options = " ".join(
+            f"{key} '{value}'" if isinstance(value, str) else f"{key} {value}"
+            for key, value in native_support_kwargs.items()
+        )
+        if iam_role:
+            sql_statement = (
+                f"COPY {table_name} "
+                f"FROM '{source_file.path}' "
+                f"IAM_ROLE '{iam_role}' "
+                f"{file_type} "
+                f"{copy_options}"
+            )
+        else:
+            aws_creds = source_file.location.hook.get_credentials()
+            sql_statement = (
```

Comment:
@pankajkoti isn't this a security issue? should we do this?

cc: @kaxil

Reply:
Which is a security issue you mean?

Reply:
btw had discussed this with @utkarsharma2 on what he meant by security issue. The implementation provides two options for authorization, one using IAM role and the other using IAM user. If IAM role is not provided, it assumes the creds are for IAM user and then uses the access_key_id and secret_access_key pair for authentication. Utkarsh feels that using this pair is a security issue and in general we should not provide this option of IAM user and only provide option of IAM role.

Reply:
sure, Yeah I am happy with that